pytorch_lightning/utilities/upgrade_checkpoint.py

```diff
@@ -0,0 +1,39 @@
+import argparse
+from shutil import copyfile
+
+import torch
+
+from pytorch_lightning.callbacks import EarlyStopping, ModelCheckpoint
+
+KEYS_MAPPING = {
+    "checkpoint_callback_best_model_score": (ModelCheckpoint, "best_model_score"),
+    "checkpoint_callback_best_model_path": (ModelCheckpoint, "best_model_path"),
+    "checkpoint_callback_best": (ModelCheckpoint, "best_model_score"),
+    "early_stop_callback_wait": (EarlyStopping, "wait_count"),
+    "early_stop_callback_patience": (EarlyStopping, "patience"),
+}
+
+
+def upgrade_checkpoint(filepath):
```

Comment:
This is great. In the future we may have to think about adding the PL version to the checkpoint. I think it will be easier to determine deprecated checkpoints based on the version rather than the keys. If we have specific functions to upgrade from one verstion to another, we may chain them together.

Reply:
yeah that's a great idea! i don't know how much we'll expect this structure to change but it would be a nice thing to consider adding

Reply:
do you mean this?
https://github.com/PyTorchLightning/pytorch-lightning/blob/4354690e55704f0c1742e5a95411b63c6055ddc7/pytorch_lightning/trainer/training_io.py#L324

Reply:
@Borda yes! I think until now, we don't do anything with it because until now we never had to convert checkpoint formats :) except the hparams thing :)